### DISCUSSION TITLE
payload type ID_V1 was not encrypted / could not decrypt payloads

### DISCUSSION THREAD

**Discussion by Baquara** [strongswan/strongswan#1322] [Q&A]
Greetings,

I am trying to connect to my company's VPN (which is an l2tp VPN), which works fine on Windows (through Sonicwall) on the same machine but not on my Linux (Arch) install.

So my credentials and PSK are not to blame here.

The Strongswan service is up and running and it manages to authenticate and initiate transactions, but it outputs the errors "payload type ID_V1 was not encrypted / could not decrypt payloads", as you can see in my `journalctl -u NetworkManager -b 0` output:

01 05:42:18 dellpc NetworkManager[508]: <info>  [1664613738.4419] vpn[0x5576cae18110,dc3e0978-9f61-4e89-bc02-e949b88d7674,"MYVPN"]: starting l2tp
 01 05:42:18 dellpc NetworkManager[508]: <info>  [1664613738.4421] audit: op="connection-activate" uuid="dc3e0978-9f61-4e89-bc02-e949b88d7674" name="MYVPN" pid=936 uid=1000 result="success"
 01 05:42:18 dellpc nm-l2tp-service[2415]: Check port 1701
 01 05:42:18 dellpc NetworkManager[2429]: Stopping strongSwan IPsec failed: starter is not running
 01 05:42:20 dellpc NetworkManager[2426]: Starting strongSwan 5.9.7 IPsec [starter]...
 01 05:42:20 dellpc ipsec_starter[2426]: Starting strongSwan 5.9.7 IPsec [starter]...
 01 05:42:20 dellpc NetworkManager[2426]: Loading config setup
 01 05:42:20 dellpc NetworkManager[2426]: Loading conn 'dc3e0978-9f61-4e89-bc02-e949b88d7674'
 01 05:42:20 dellpc ipsec_starter[2426]: Loading config setup
 01 05:42:20 dellpc ipsec_starter[2426]: Loading conn 'dc3e0978-9f61-4e89-bc02-e949b88d7674'
 01 05:42:20 dellpc ipsec_starter[2437]: Attempting to start charon...
 01 05:42:20 dellpc charon[2439]: 00[DMN] Starting IKE charon daemon (strongSwan 5.9.7, Linux 5.19.11-arch1-1, x86_64)
 01 05:42:20 dellpc charon[2439]: 00[CFG] PKCS11 module '<name>' lacks library path
 01 05:42:20 dellpc charon[2439]: 00[KNL] unable to create IPv4 ring table rule
 01 05:42:20 dellpc charon[2439]: 00[KNL] unable to create IPv6 ring table rule
 01 05:42:20 dellpc charon[2439]: 00[CFG] attr-sql plugin: database URI not set
 01 05:42:20 dellpc charon[2439]: 00[NET] using forecast interface wlan0
 01 05:42:20 dellpc charon[2439]: 00[CFG] joining forecast multicast groups: 224.0.0.1,224.0.0.22,224.0.0.251,224.0.0.252,239.255.255.250
 01 05:42:20 dellpc charon[2439]: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
 01 05:42:20 dellpc charon[2439]: 00[CFG]   loaded ca certificate "C=CH, O=strongSwan, CN=strongSwan Root CA" from '/etc/ipsec.d/cacerts/strongswanCert.pem'
 01 05:42:20 dellpc charon[2439]: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
 01 05:42:20 dellpc charon[2439]: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
 01 05:42:20 dellpc charon[2439]: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
 01 05:42:20 dellpc charon[2439]: 00[CFG] loading crls from '/etc/ipsec.d/crls'
 01 05:42:20 dellpc charon[2439]: 00[CFG] loading secrets from '/etc/ipsec.secrets'
 01 05:42:20 dellpc charon[2439]: 00[CFG] loading secrets from '/etc/ipsec.d/ipsec.nm-l2tp.secrets'
 01 05:42:20 dellpc charon[2439]: 00[CFG]   loaded IKE secret for %any
 01 05:42:20 dellpc charon[2439]: 00[CFG] sql plugin: database URI not set
 01 05:42:20 dellpc charon[2439]: 00[CFG] opening triplet file /etc/ipsec.d/triplets.dat failed: No such file or directory
 01 05:42:20 dellpc charon[2439]: 00[CFG] loaded 0 RADIUS server configurations
 01 05:42:20 dellpc charon[2439]: 00[CFG] HA config misses local/remote address
 01 05:42:20 dellpc charon[2439]: 00[CFG] no script for ext-auth script defined, disabled
 01 05:42:20 dellpc charon[2439]: 00[LIB] loaded plugins: charon ldap pkcs11 aesni aes des rc2 sha2 sha3 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs12 pgp dnskey sshkey pem openssl pkcs>
 01 05:42:20 dellpc charon[2439]: 00[LIB] dropped capabilities, running as uid 0, gid 0
 01 05:42:20 dellpc charon[2439]: 00[JOB] spawning 16 worker threads
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[IKE] installed bypass policy for 192.168.1.0/24
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[IKE] installed bypass policy for ::1/128
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[KNL] policy already exists, try to update it
 01 05:42:20 dellpc charon[2439]: 05[IKE] installed bypass policy for fe80::/64
 01 05:42:20 dellpc ipsec_starter[2437]: charon (2439) started after 20 ms
 01 05:42:20 dellpc charon[2439]: 08[CFG] received stroke: add connection 'dc3e0978-9f61-4e89-bc02-e949b88d7674'
 01 05:42:20 dellpc charon[2439]: 08[CFG] added configuration 'dc3e0978-9f61-4e89-bc02-e949b88d7674'
 01 05:42:21 dellpc charon[2439]: 10[CFG] rereading secrets
 01 05:42:21 dellpc charon[2439]: 10[CFG] loading secrets from '/etc/ipsec.secrets'
 01 05:42:21 dellpc charon[2439]: 10[CFG] loading secrets from '/etc/ipsec.d/ipsec.nm-l2tp.secrets'
 01 05:42:21 dellpc charon[2439]: 10[CFG]   loaded IKE secret for %any
 01 05:42:21 dellpc charon[2439]: 13[CFG] received stroke: initiate 'dc3e0978-9f61-4e89-bc02-e949b88d7674'
 01 05:42:21 dellpc charon[2439]: 16[IKE] initiating Main Mode IKE_SA dc3e0978-9f61-4e89-bc02-e949b88d7674[1] to xxx.y.zz.nnn
 01 05:42:21 dellpc charon[2439]: 16[IKE] initiating Main Mode IKE_SA dc3e0978-9f61-4e89-bc02-e949b88d7674[1] to xxx.y.zz.nnn
 01 05:42:21 dellpc charon[2439]: 16[ENC] generating ID_PROT request 0 [ SA V V V V V ]
 01 05:42:21 dellpc charon[2439]: 16[NET] sending packet: from 192.168.1.19[500] to xxx.y.zz.nnn[500] (240 bytes)
 01 05:42:21 dellpc charon[2439]: 07[NET] received packet: from xxx.y.zz.nnn[500] to 192.168.1.19[500] (116 bytes)
 01 05:42:21 dellpc charon[2439]: 07[ENC] parsed ID_PROT response 0 [ SA V V ]
 01 05:42:21 dellpc charon[2439]: 07[IKE] received Sonicwall 7 vendor ID
 01 05:42:21 dellpc charon[2439]: 07[IKE] received NAT-T (RFC 3947) vendor ID
 01 05:42:21 dellpc charon[2439]: 07[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_1024
 01 05:42:21 dellpc charon[2439]: 07[ENC] generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
 01 05:42:21 dellpc charon[2439]: 07[NET] sending packet: from 192.168.1.19[500] to xxx.y.zz.nnn[500] (268 bytes)
 01 05:42:21 dellpc charon[2439]: 06[NET] received packet: from xxx.y.zz.nnn[500] to 192.168.1.19[500] (300 bytes)
 01 05:42:21 dellpc charon[2439]: 06[ENC] parsed ID_PROT response 0 [ KE NAT-D NAT-D No V V V ]
 01 05:42:21 dellpc charon[2439]: 06[IKE] received Sonicwall a vendor ID
 01 05:42:21 dellpc charon[2439]: 06[IKE] received XAuth vendor ID
 01 05:42:21 dellpc charon[2439]: 06[IKE] received DPD vendor ID
 01 05:42:21 dellpc charon[2439]: 06[IKE] local host is behind NAT, sending keep alives
 01 05:42:21 dellpc charon[2439]: 06[ENC] generating ID_PROT request 0 [ ID HASH ]
 01 05:42:21 dellpc charon[2439]: 06[NET] sending packet: from 192.168.1.19[4500] to xxx.y.zz.nnn[4500] (92 bytes)
 01 05:42:21 dellpc charon[2439]: 14[NET] received packet: from xxx.y.zz.nnn[4500] to 192.168.1.19[4500] (92 bytes)
 01 05:42:21 dellpc charon[2439]: 14[IKE] queueing TRANSACTION request as tasks still active
 01 05:42:25 dellpc charon[2439]: 11[IKE] sending retransmit 1 of request message ID 0, seq 3
 01 05:42:25 dellpc charon[2439]: 11[NET] sending packet: from 192.168.1.19[4500] to xxx.y.zz.nnn[4500] (92 bytes)
 01 05:42:25 dellpc charon[2439]: 10[NET] received packet: from xxx.y.zz.nnn[4500] to 192.168.1.19[4500] (104 bytes)
 01 05:42:25 dellpc charon[2439]: 10[ENC] payload type ID_V1 was not encrypted
 01 05:42:25 dellpc charon[2439]: 10[ENC] could not decrypt payloads
 01 05:42:25 dellpc charon[2439]: 10[IKE] integrity check failed
 01 05:42:25 dellpc charon[2439]: 10[ENC] generating INFORMATIONAL_V1 request 2196094742 [ HASH N(INVAL_HASH) ]
 01 05:42:25 dellpc charon[2439]: 10[NET] sending packet: from 192.168.1.19[4500] to xxx.y.zz.nnn[4500] (92 bytes)
 01 05:42:25 dellpc charon[2439]: 10[IKE] ID_PROT response with message ID 0 processing failed
 01 05:42:28 dellpc NetworkManager[508]: <warn>  [1664613748.4617] vpn[0x5576cae18110,dc3e0978-9f61-4e89-bc02-e949b88d7674,"MYVPN"]: failed to connect: 'O tempo limite foi alcançado'
 01 05:42:31 dellpc NetworkManager[2471]: Stopping strongSwan IPsec...
 01 05:42:31 dellpc charon[2439]: 00[DMN] SIGINT received, shutting down
 01 05:42:31 dellpc charon[2439]: 00[IKE] destroying IKE_SA in state CONNECTING with notification
 01 05:42:31 dellpc NetworkManager[2465]: initiating Main Mode IKE_SA dc3e0978-9f61-4e89-bc02-e949b88d7674[1] to xxx.y.zz.nnn
 01 05:42:31 dellpc NetworkManager[2465]: generating ID_PROT request 0 [ SA V V V V V ]
 01 05:42:31 dellpc NetworkManager[2465]: sending packet: from 192.168.1.19[500] to xxx.y.zz.nnn[500] (240 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: received packet: from xxx.y.zz.nnn[500] to 192.168.1.19[500] (116 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: parsed ID_PROT response 0 [ SA V V ]
 01 05:42:31 dellpc NetworkManager[2465]: received Sonicwall 7 vendor ID
 01 05:42:31 dellpc NetworkManager[2465]: received NAT-T (RFC 3947) vendor ID
 01 05:42:31 dellpc NetworkManager[2465]: selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_1024
 01 05:42:31 dellpc NetworkManager[2465]: generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
 01 05:42:31 dellpc NetworkManager[2465]: sending packet: from 192.168.1.19[500] to xxx.y.zz.nnn[500] (268 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: received packet: from xxx.y.zz.nnn[500] to 192.168.1.19[500] (300 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: parsed ID_PROT response 0 [ KE NAT-D NAT-D No V V V ]
 01 05:42:31 dellpc NetworkManager[2465]: received Sonicwall a vendor ID
 01 05:42:31 dellpc NetworkManager[2465]: received XAuth vendor ID
 01 05:42:31 dellpc NetworkManager[2465]: received DPD vendor ID
 01 05:42:31 dellpc NetworkManager[2465]: local host is behind NAT, sending keep alives
 01 05:42:31 dellpc NetworkManager[2465]: generating ID_PROT request 0 [ ID HASH ]
 01 05:42:31 dellpc NetworkManager[2465]: sending packet: from 192.168.1.19[4500] to xxx.y.zz.nnn[4500] (92 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: received packet: from xxx.y.zz.nnn[4500] to 192.168.1.19[4500] (92 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: queueing TRANSACTION request as tasks still active
 01 05:42:31 dellpc NetworkManager[2465]: sending retransmit 1 of request message ID 0, seq 3
 01 05:42:31 dellpc NetworkManager[2465]: sending packet: from 192.168.1.19[4500] to xxx.y.zz.nnn[4500] (92 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: received packet: from xxx.y.zz.nnn[4500] to 192.168.1.19[4500] (104 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: payload type ID_V1 was not encrypted
 01 05:42:31 dellpc NetworkManager[2465]: could not decrypt payloads
 01 05:42:31 dellpc NetworkManager[2465]: integrity check failed
 01 05:42:31 dellpc NetworkManager[2465]: generating INFORMATIONAL_V1 request 2196094742 [ HASH N(INVAL_HASH) ]
 01 05:42:31 dellpc NetworkManager[2465]: sending packet: from 192.168.1.19[4500] to xxx.y.zz.nnn[4500] (92 bytes)
 01 05:42:31 dellpc NetworkManager[2465]: ID_PROT response with message ID 0 processing failed
 01 05:42:31 dellpc NetworkManager[2465]: destroying IKE_SA in state CONNECTING with notification
 01 05:42:31 dellpc NetworkManager[2465]: establishing connection 'dc3e0978-9f61-4e89-bc02-e949b88d7674' failed
 01 05:42:31 dellpc charon[2439]: 00[IKE] uninstalling bypass policy for 192.168.1.0/24
 01 05:42:31 dellpc charon[2439]: 00[IKE] uninstalling bypass policy for ::1/128
 01 05:42:31 dellpc charon[2439]: 00[IKE] uninstalling bypass policy for fe80::/64
 01 05:42:31 dellpc ipsec_starter[2437]: child 2439 (charon) has quit (exit code 0)
 01 05:42:31 dellpc ipsec_starter[2437]:
 01 05:42:31 dellpc ipsec_starter[2437]: charon stopped after 200 ms
 01 05:42:31 dellpc ipsec_starter[2437]: ipsec starter stopped
 01 05:42:31 dellpc nm-l2tp-service[2415]: Could not establish IPsec connection.
 01 05:42:31 dellpc nm-l2tp-service[2415]: g_dbus_method_invocation_take_error: assertion 'error != NULL' failed
 01 05:42:32 dellpc NetworkManager[508]: <info>  [1664613752.8007] audit: op="statistics" interface="wlan0" ifindex=3 args="0" pid=936 uid=1000 result="success"

My settings follows:

![image](https://user-images.githubusercontent.com/9168959/193402044-da632c9a-f9d9-476a-bfae-6edf5fad2f0a.png)

![image](https://user-images.githubusercontent.com/9168959/193402054-e493e42a-fd80-495b-a85f-098e48e077ed.png)

**Comment by tobiasbrunner** [comment]
> So my credentials and PSK are not to blame here.

According to the log, it's most likely an incorrect PSK, though.

**Reply by Baquara** [reply]
> > So my credentials and PSK are not to blame here.
> 
> According to the log, it's most likely an incorrect PSK, though.

The same PSK is being used on other systems and they work just fine, while it does not in mine. I am pretty positive it is something else...

**Reply by Thermi** [reply]
I had weird issues like that at a location too. After switching IP addresses (yes really just that) the problem disappeared.
